pkg/queryfrontend/labels_codec.go

```diff
@@ -197,7 +197,7 @@ func (c labelsCodec) DecodeResponse(ctx context.Context, r *http.Response, req q
 		body, _ := ioutil.ReadAll(r.Body)
 		return nil, httpgrpc.Errorf(r.StatusCode, string(body))
 	}
-	log, ctx := spanlogger.New(ctx, "ParseQueryResponse") //nolint:ineffassign,staticcheck
```

Comment:
I think this was intentional. Let's say if someone tried to create another span in this function, they need to use the updated `ctx` coming from `spanlogge.New` but they might miss that.

I am not sure though so I'll wait for someone else to review.

Reply:
Maybe we can add a proper comment to give a hint to the future developers and remove `nolint` statement and the variable. This is definitely intentional.